compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -386,6 +366,28 @@ object Completion {
         !name.isConstructorName && name.toTermName.info.kind == SimpleNameKind
       def isStable = true
     }
+
+    extension (scope: CompletionScope)
+      private def withDenots(denotations: Seq[SingleDenotation], name: Name)(using Context): CompletionScope = {
+        val denots = denotations.filter(denot => include(denot.symbol, name))
+        val shortName = name.stripModuleClassSuffix
+
+        if denots.nonEmpty then
+          CompletionScope(scope.nameToDenots + (shortName -> denots.toList))
+        else
+          scope
+      }
+
+    extension (scope: CompletionScope.type)
```

Comment:
That would require making `CompletionScope` an inner class of `CompletionBuffer` so that it has easy access to the name filter defined in it but maybe that's not a problem